### ISSUE TITLE
Waterway's name render is inconsistent. 

### ISSUE THREAD

**Issue by DaveF63** [gravitystorm/openstreetmap-carto#2411]
Hi
https://www.openstreetmap.org/#map=17/51.39099/-2.34528

There's an inconsistency between the rendering of the waterway's name for rivers & canals. Is this intentional? Would it be better if they occurred at the same frequency?
Personally I think rivers render slightly too often & canals not frequently enough. Maybe a compromise to somewhere in the middle?

**Comment by pnorman** [comment]
looks like rivers have `text-spacing: 400;` and canals have nothing set

**Comment by DaveF63** [comment]
Is that in pixels? Maybe 600/800 for both?

**Comment by kocio-pl** [comment]
@DaveF63 Could you try to test these values at least initially?

**Comment by DaveF63** [comment]
As an end user how would i go about that? Is there a carto sandbox which renders amendments, as the dev sandbox doesn't do it.

**Comment by SomeoneElseOSM** [comment]
It doesn't cover setting up a test version of OSM Carto from scratch, but I wrote a bit about text spacing in a [recent diary entry](http://www.openstreetmap.org/user/SomeoneElse/diary/38988).  That's talking about long distance footpaths, and is not based on the current "standard" style, but those comments should be applicable to that.

**Comment by kocio-pl** [comment]
@DaveF63 Now that we have a Docker-based testing environment, you can try test rendering yourself, see:

https://github.com/gravitystorm/openstreetmap-carto/blob/master/DOCKER.md
https://ircama.github.io/osm-carto-tutorials/

**Comment by Adamant36** [comment]
Here's some examples at different zoom and pixel levels. Unfortunately its only the name of river because I realized after doing it that the canal was miss tagged as a creek. So it didn't change at throughout the process. I figured id at least add examples for the river though since they are going to have the same name spacing anyway. If its an issue, I can redo it. 


First we have 600
![600 100 to 500 zoom](https://user-images.githubusercontent.com/30259065/34972981-e7f5bb08-fa39-11e7-9ed1-52d9c84435da.png)
![600 300 to 1000 zoom](https://user-images.githubusercontent.com/30259065/34972997-fdbece7a-fa39-11e7-97ca-27714348ffb3.png)
![600 500 to 1000 zoom](https://user-images.githubusercontent.com/30259065/34973001-0937908e-fa3a-11e7-907a-ef97f71219ee.png)
Now 700
![700 100 to 500 zoom](https://user-images.githubusercontent.com/30259065/34973029-3904a5ae-fa3a-11e7-9c0d-fdaca72ce294.png)
![700 300 to 1000 zoom](https://user-images.githubusercontent.com/30259065/34973044-447aba9a-fa3a-11e7-908c-71a94b56a863.png)
![700 500 2000](https://user-images.githubusercontent.com/30259065/34973059-50ae3378-fa3a-11e7-96c8-22382d3ed5f8.png)
And finally 800
![800 100 to 500 zoom](https://user-images.githubusercontent.com/30259065/34973071-6559d282-fa3a-11e7-960d-a34baa6ad7ad.png)
![800 300 1000 zoom](https://user-images.githubusercontent.com/30259065/34973080-79ff5540-fa3a-11e7-8c78-3b91271e4e34.png)
![800 500 to 2000 zoom](https://user-images.githubusercontent.com/30259065/34973089-883e08fe-fa3a-11e7-8243-fc1a46e3466a.png)

I didn't do 500 because it was not within the range the original poster mentioned @pnorman mentioned and it seemed to similar to 400 when I looked at it, but I can do that to if anyone wants to see it. Personally, I think 600 is the most even looking, although they all seem to have their pluses and minuses at different zoom levels. 700 seems to have a render problem at 300/1000 zoom, where as 800 has one at 500/2000 zoom. Although that could just be because of the place in the river that the name is located at. Looking back, I probably could have found somewhere straighter to try it out, but it was the only place I knew of close to a canal. Plus, rivers are not always that straight..So...Maybe instead of picking one pixel width for every zoom level, it could be variable.

**Comment by kocio-pl** [comment]
I think the most important would be not to fix on one example, but to test at least few different waterways, with different shapes and other features. 

It can be not clear from the start which features are relevant to labels rendering (you might get some that are not important even if you think they are, and the other way around), so making some random choices helps to find interesting cases.

Would you like to try?

**Comment by Adamant36** [comment]
@kocio-pl Good call. It looks like creeks have the same issue. So I'll probably tweak around with them to. Do you think the label spacing should be the same for all types of waterways and is there a preferred zoom level to test things on so I don't have to upload a bunch of screenshots for every spacing I test?

**Comment by kocio-pl** [comment]
My working theory would be that streams are smaller and shorter and there are not too many surrounding objects, so they might need more labels than rivers and canals, but this is one of the assumptions that should be checked. I have no idea what zoom levels and range should be tested - the more, the better, until you find some patterns.

What works for me is to start with the objects/places I know and then trying to find something known, strange, different (for example on the other continent) or unusual in some other way and then I try as much as I can handle.
